flocker/acceptance/endtoend/test_dataset.py

```diff
@@ -102,3 +109,69 @@ def not_exists():
             return deleted
         created.addCallback(delete_dataset)
         return created
+
+    @require_moving_backend
+    @run_test_with(async_runner(timeout=timedelta(minutes=6)))
+    @require_cluster(2)
+    def test_dataset_move_from_dead_node(self, cluster):
+        """
+        A dataset can be moved from one node to another.
+
+        All attributes, including the maximum size, are preserved.
+        """
+        api = get_backend_api(self, cluster.cluster_uuid)
```

Comment:
The current implementation of get_backend_api only constructs aws backends (sorry, I wasn't sure if this api would be in general useful in acceptance tests so I didn't put the work in upfront). If we are claiming OpenStack in the release notes we should probably test on rackspace as well.

Shouldn't be too much work - something like making _DEFAULT_BACKENDS public from flocker/node/script.py:459 and using it in get_backend_api to construct backends of any type.